Source/Diagnostics/ReducedDiags/ParticleExtrema.cpp

```diff
@@ -0,0 +1,452 @@
+
+        // get mass (Real)
+        auto m = myspc.getMass();
+        auto is_photon = myspc.AmIA<PhysicalSpecies::photon>();
+        if ( is_photon ) {
+            m = PhysConst::m_e;
+        }
+
+        using PType = typename WarpXParticleContainer::SuperParticleType;
+
+        // xmin
+#if (defined WARPX_DIM_RZ)
+        Real xmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(0)*std::cos(p.rdata(PIdx::theta)); });
+        ParallelDescriptor::ReduceRealMin(xmin);
+#else
+        Real xmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(0); });
+        ParallelDescriptor::ReduceRealMin(xmin);
+#endif
+
+        // xmax
+#if (defined WARPX_DIM_RZ)
+        Real xmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(0)*std::cos(p.rdata(PIdx::theta)); });
+        ParallelDescriptor::ReduceRealMax(xmax);
+#else
+        Real xmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(0); });
+        ParallelDescriptor::ReduceRealMax(xmax);
+#endif
+
+        // ymin
+#if (defined WARPX_DIM_RZ)
+        Real ymin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(0)*std::sin(p.rdata(PIdx::theta)); });
+        ParallelDescriptor::ReduceRealMin(ymin);
+#elif (defined WARPX_DIM_XZ)
+        Real ymin = 0.0_rt;
+#else
+        Real ymin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(1); });
+        ParallelDescriptor::ReduceRealMin(ymin);
+#endif
+
+        // ymax
+#if (defined WARPX_DIM_RZ)
+        Real ymax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(0)*std::sin(p.rdata(PIdx::theta)); });
+        ParallelDescriptor::ReduceRealMax(ymax);
+#elif (defined WARPX_DIM_XZ)
+        Real ymax = 0.0_rt;
+#else
+        Real ymax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(1); });
+        ParallelDescriptor::ReduceRealMax(ymax);
+#endif
+
+        // zmin
+        Real zmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(index_z); });
+        ParallelDescriptor::ReduceRealMin(zmin);
+
+        // zmax
+        Real zmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.pos(index_z); });
+        ParallelDescriptor::ReduceRealMax(zmax);
+
+        // uxmin
+        Real uxmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.rdata(PIdx::ux); });
+        ParallelDescriptor::ReduceRealMin(uxmin);
+
+        // uxmax
+        Real uxmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+        { return p.rdata(PIdx::ux); });
+        ParallelDescriptor::ReduceRealMax(uxmax);
+
+        // uymin
```

Comment:
I was wondering: are all those calls to `ReduceMin` and `ReduceMax` an issue? Would it be better to group all them in a `reduce_op.eval` loop ?
(if this is the case we might do that in another PR)

Reply:
Great idea! They are probably not a severe issue per se, but grouping them would definitely be a nice follow-up PR to speed up the diagnostics (latency and performance at scale).